test/libsolidity/util/TestFileParser.cpp

```diff
@@ -192,51 +199,121 @@ Parameter TestFileParser::parseParameter()
 
 tuple<bytes, ABIType, string> TestFileParser::parseABITypeLiteral()
 {
-	try
+	enum class DeclaredAlignment {
+		Left,
+		Right,
+		None,
+	};
+
+	auto alignLeft = [](bytes _bytes) -> bytes
 	{
-		u256 number{0};
-		ABIType abiType{ABIType::None, ABIType::AlignRight, 0};
-		string rawString;
+		return _bytes + bytes(32 - _bytes.size(), 0);
+	};
 
-		if (accept(Token::Sub))
+	auto alignRight = [](bytes _bytes) -> bytes
+	{
+		return bytes(32 - _bytes.size(), 0) + _bytes;
+	};
+
+	auto applyAlign = [&](DeclaredAlignment _modifier, bytes _bytes) -> pair<bytes, ABIType::Align>
+	{
+		if (_modifier == DeclaredAlignment::Right)
+			return make_pair(alignRight(_bytes), ABIType::AlignRight);
```

Comment:
You could probably inline ``alignRight`` (and ``alignLeft`` below). I also wonder whether skipping the whole ``DeclaredAlignment`` business and using a capture-less ``alignment`` lambda instead, supplemented by a ``hasAlignModifier`` flag for the ``alignment != Alignment::None`` check below. Probably fine either way, though.
Also you could pass in a reference to ``ABIType`` and set its ``align`` member in the lambda instead of returning a pair - that might spare you reassigning ``ABIType::align`` over and over below. For that matter it could also just be a ref to ``bytes`` to which the padding is appended/prepended.